src/index.js

```diff
@@ -1,112 +1,102 @@
-                } else if (varName == null) {
-                    cb(null);
-                } else {
-                    cb(null, new ExternalModule(varName, 'var', modulePath));
+        // Make the external modules available to other plugins
+        compiler.hooks.compilation.tap(pluginName, compilation => {
+            compilation.hooks.beforeModuleAssets.tap(pluginName, () => {
+                for (const [name, cdnConfig] of Object.entries(this.modulesFromCdn)) {
+                    compilation.addChunkInGroup(name);
+                    const chunk = compilation.addChunk(name);
+                    chunk.files.add(cdnConfig.url);
                 }
             });
+
+            // Optionally, update the HtmlWebpackPlugin assets
+            const isUsingHtmlWebpackPlugin = compiler.options.plugins.some(x => x instanceof HtmlWebpackPlugin);
+            if (isUsingHtmlWebpackPlugin) {
+                const hooks = HtmlWebpackPlugin.getHooks(compilation);
+                hooks.beforeAssetTagGeneration.tapAsync(pluginName, (htmlPluginData, callback) => {
+                    const {assets} = htmlPluginData;
+                    const scripts = Object.values(this.modulesFromCdn).map(moduleFromCdn => moduleFromCdn.url);
+                    assets.js = scripts.concat(assets.js);
+                    return callback(null, htmlPluginData);
+                });
+            }
         });
     }
 
     async addModule(contextPath, modulePath, {env}) {
         const isModuleExcluded = this.exclude.includes(modulePath) ||
             (this.only && !this.only.includes(modulePath));
-        if (isModuleExcluded) {
-            return false;
-        }
+        if (isModuleExcluded) return false;
 
         const moduleName = modulePath.match(moduleRegex)[1];
-        const {pkg: {version, peerDependencies}} = await readPkgUp({cwd: resolvePkg(moduleName, {cwd: contextPath})});
+        const {packageJson: {version, peerDependencies}} = await readPkgUp({cwd: resolvePkg(moduleName, {cwd: contextPath})});
 
         const isModuleAlreadyLoaded = Boolean(this.modulesFromCdn[modulePath]);
         if (isModuleAlreadyLoaded) {
             const isSameVersion = this.modulesFromCdn[modulePath].version === version;
-            if (isSameVersion) {
-                return this.modulesFromCdn[modulePath].var;
-            }
-
-            return false;
+            return isSameVersion ? this.modulesFromCdn[modulePath].var : false;
         }
 
         const cdnConfig = await this.resolver(modulePath, version, {env});
 
         if (cdnConfig == null) {
-            if (this.verbose) {
-                console.log(`❌ '${modulePath}' couldn't be found, please add it to https://github.com/mastilver/module-to-cdn/blob/master/modules.json`);
-            }
+            if (this.verbose) console.log(`❌ '${modulePath}' couldn't be found, please add it to https://github.com/mastilver/module-to-cdn/blob/master/modules.json`);
```

Comment:
Same as above, and in this case the `console.log` line is already long so it difficults reading the whole line on small or split screens.